Practice/Practice.Repository/HumanResourcesRepository.cs

```diff
@@ -31,5 +33,70 @@ public async Task<List<ViewModels.Employee>> GetEmployees()
 
             return employees;
         }
+
+        public async Task<ViewModels.Employee> GetEmployeeById(int id)
+        {
+            if (id == 0)
+                return null;
+
+            var employee = await context.vEmployee.FirstOrDefaultAsync(e => e.BusinessEntityID == id);
+            if (employee == null)
+                return null;
+
+            return new ViewModels.Employee()
+            {
+                Id = employee.BusinessEntityID,
+                Address = employee.AddressLine1,
+                FirstName = employee.FirstName,
+                LastName = employee.LastName,
+                City = employee.City,
+                PhoneNumber = employee.PhoneNumber,
+                PostalCode = employee.PostalCode
+            };
+        }
+
+        public async Task UpdateEmployee(ViewModels.Employee employee)
+        {
+            if (employee == null || employee.Id == 0)
+                return;
+
+            var person = await context.Person.FirstOrDefaultAsync(p => p.BusinessEntityID == employee.Id);
+            if (person == null)
+                return;
+
+            // First and Last name
+            person.FirstName = employee.FirstName;
+            person.LastName = employee.LastName;
+
+            // Phone
+            var oldPhone = person.PersonPhone.FirstOrDefault();
+            if (oldPhone != null)
+            {
+                person.PersonPhone.Remove(oldPhone);
```

Comment:
Tego nie rozumiem, po co usuwać wpis przy edycji? Wystarczy nadpisać pole PhoneNumber - można to zrobić najprościej: person.PersonPhone.First().PhoneNumber = employee.PhoneNumber;

Reply:
Kluczem głównym w PersonPhone jest złączenie pól BusinessEntityID-PhoneNumber-PhoneNumberTypeID i gdy chciałem zmienić PhoneNumber dostawałem wyjątek, że nie można modyfikować klucza głównego, więc musiałem usunąć rekord, stworzyć kopie, zmienić numer i dodać nowy do bazy.